daemon/algod/api/algod.oas2.json

```diff
@@ -3964,24 +4011,37 @@
         }
       }
     },
-    "SimulationResponse": {
+    "SimulateResponse": {
       "description": "Result of a transaction group simulation.",
       "tags": [
         "experimental"
       ],
       "schema": {
         "type": "object",
         "required": [
-          "failure-message",
-          "missing-signatures"
+          "version",
+          "last-round",
+          "txn-groups",
+          "would-succeed"
         ],
         "properties": {
-          "failure-message": {
-            "description": "\\[fm\\] Failure message, if the transaction would have failed during a live broadcast.",
-            "type": "string"
+          "version": {
+            "description": "The version of this response object.",
+            "type": "integer"
+          },
+          "last-round": {
+            "description": "The round immediately preceding this simulation. State changes through this round were used to run this simulation.",
+            "type": "integer"
+          },
+          "txn-groups": {
+            "description": "A result object for each transaction group that was simulated.",
+            "type": "array",
+            "items": {
+              "$ref": "#/definitions/SimulateTransactionGroupResult"
+            }
           },
-          "missing-signatures": {
-            "description": "\\[ms\\] Whether any transactions would have failed during a live broadcast because they were missing signatures.",
+          "would-succeed": {
+            "description": "Indicates whether the simulated transactions would have succeeded during an actual submission.",
```

Comment:
I suppose this is true if _all_ of the transaction groups would have succeeded? I think I'd just drop it, but I don't feel strongly.

Reply:
The motivation behind "would-succeed" at this level was to provide a dead simple way of figuring out if your transactions would succeed if you sent them to the real submission endpoint. Yes, this is redundant information, since you could deduce the same thing by checking that no TxnGroup has a failure message and no Txn is missing signatures.

But in the future you may have to check more things, like no additional foreign resources were needed and you didn't exceed the standard opcode budget. True, these features would likely be opt-in so they're less likely to confuse people, but still think there's a benefit to summarizing all the possible reasons for failure in a simple boolean.

Reply:
I'm ok with this, but I'm actually so convinced that I think it should exist on individual txn-groups as well (we can add in a later PR, since it will be an addition).  The idea that there will be multiple ways that we might allow a group to "succeed" even though they would not actually succeed on chain is what convinces me. And, I think there will be a good use for simulation in unit tests where you want to confirm that, say, txgroups 1, 2 , 3 and 5 would succeed, but 4 would not.  That is, the convenience you're offering is good at the txgroup level, because you may want some of the groups to fail (you are testing that they will, in fact). But that the other groups are perfectly correct.

Reply:
Those are good points. I'll admit I haven't given much thought to how multiple groups would be handled, but your ideas make sense.